py/selenium/webdriver/chrome/options.py

```diff
@@ -19,158 +19,14 @@
-            encoded_extensions.append(base64.b64encode(file_.read()).decode('UTF-8'))
-
-            file_.close()
-        return encoded_extensions + self._extensions
-
-    def add_extension(self, extension):
-        """
-        Adds the path to the extension to a list that will be used to extract it
-        to the ChromeDriver
-
-        :Args:
-         - extension: path to the \\*.crx file
-        """
-        if extension:
-            extension_to_add = os.path.abspath(os.path.expanduser(extension))
-            if os.path.exists(extension_to_add):
-                self._extension_files.append(extension_to_add)
-            else:
-                raise IOError("Path to the extension doesn't exist")
-        else:
-            raise ValueError("argument can not be null")
-
-    def add_encoded_extension(self, extension):
-        """
-        Adds Base64 encoded string with extension data to a list that will be used to extract it
-        to the ChromeDriver
-
-        :Args:
-         - extension: Base64 encoded string with extension data
-        """
-        if extension:
-            self._extensions.append(extension)
-        else:
-            raise ValueError("argument can not be null")
-
-    @property
-    def experimental_options(self):
-        """
-        :Returns: A dictionary of experimental options for chrome
-        """
-        return self._experimental_options
-
-    def add_experimental_option(self, name, value):
-        """
-        Adds an experimental option which is passed to chrome.
-
-        :Args:
-          name: The experimental option name.
-          value: The option value.
-        """
-        self._experimental_options[name] = value
-
-    @property
-    def headless(self):
-        """
-        :Returns: True if the headless argument is set, else False
-        """
-        return '--headless' in self._arguments
-
-    @headless.setter
-    def headless(self, value):
-        """
-        Sets the headless argument
-
-        :Args:
-          value: boolean value indicating to set the headless option
-        """
-        args = {'--headless'}
-        if value is True:
-            self._arguments.extend(args)
-        else:
-            self._arguments = list(set(self._arguments) - args)
-
-    def to_capabilities(self):
-        """
-        Creates a capabilities with all the options that have been set
-
-        :Returns: A dictionary with everything
-        """
-        caps = self._caps
-        chrome_options = self.experimental_options.copy()
-        chrome_options["extensions"] = self.extensions
-        if self.binary_location:
-            chrome_options["binary"] = self.binary_location
-        chrome_options["args"] = self.arguments
-        if self.debugger_address:
-            chrome_options["debuggerAddress"] = self.debugger_address
-
-        caps[self.KEY] = chrome_options
-
-        return caps
+        super(Options, self).__init__(self.KEY)
```

Comment:
You're right that each vendor will have different KEY, but _to_capabilities_ is relying on the KEY to generate capabilities, so I thought it's better to put that method in ChromiumOptions to avoid duplication.